efs.tf

```diff
@@ -1,29 +1,84 @@
-resource "aws_efs_file_system" "this" {
+resource "aws_efs_file_system" "file_system" {
   creation_token = var.name
-  tags = merge(
-    var.tags,
-    {
-      Name = var.name
-  })
+
+  encrypted        = true
+  performance_mode = "generalPurpose"
+  throughput_mode  = "bursting"
+
+  tags = {
+    Name        = "${var.name}-eks-efs"
+    Environment = var.name
+  }
 }
 
-resource "aws_efs_mount_target" "this" {
-  count           = length(var.subnet_ids)
-  file_system_id  = aws_efs_file_system.this.id
+resource "aws_efs_mount_target" "mount_target" {
+  count = length(var.subnet_ids)
+
+  file_system_id  = aws_efs_file_system.file_system.id
   subnet_id       = var.subnet_ids[count.index]
-  security_groups = var.security_group_ids
+  security_groups = [aws_security_group.mount_target_security_group.id]
 }
 
-resource "kubernetes_storage_class_v1" "this" {
+
+resource "aws_security_group" "mount_target_security_group" {
+  vpc_id = var.vpc_id
+
+  ingress {
+    from_port   = 2049
+    to_port     = 2049
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr_block]
+  }
+
+  tags = {
+    Name        = "${var.name}-eks-efs-sg"
+    Environment = var.name
+  }
+}
+
+resource "kubernetes_storage_class" "efs_sc" {
   metadata {
-    name = var.name
+    name = "efs-sc"
   }
+
   storage_provisioner = "efs.csi.aws.com"
-  parameters = {
-    provisioningMode = "efs-ap"
-    fileSystemId     = aws_efs_file_system.this.id
-    directoryPerms   = "777"
-    gidRangeStart    = "1000"    # optional
-    gidRangeEnd      = "7000000" # optional
+}
+
+resource "kubernetes_persistent_volume" "efs_pvc" {
```

Comment:
We can move `kubernetes_persistent_volume` and `kubernetes_persistent_volume_claim` to customer side